### PR TITLE
Feature/in a table

### PR DESCRIPTION
**Pull request by emclaug2**
#### Changes proposed in this Pull Request:
- Form validation in a table example 

- There's currently some weird behavior with the table; I've never used a MUI Table before, if you hover over the 2nd row there's some hover-state styles that appear.  These styles are only applied to the odd-numbered rows & may be an issue with the theme.

### CODE REVIEW

**Comment by jeffvg** [comment]
looks good but yeah has that one weird hover state. Looks like there is a prop for it..
https://material-ui.com/api/table-row/

**Comment by jeffvg** [comment]
looks like that second row inherits different background color. Might want to use data elements other than power xpert blue

**Comment by ektasawant** [comment]
Looks good 👍 .

Few comments from my side: 
1. Note is missing below table `Remember that in a real application ...`
2. Table width is not matching with design.
3. In mobile color of `#1, #2 and #3` is gray-500
4. Can we remove divider prop from last column?

**Comment by huayunh** [comment]
image
Number cells shall be right-aligned. Text cells shall stay left-aligned



image
What is this white space on the bottom for?

**Comment by emclaug2** [comment]
> looks like that second row inherits different background color. Might want to use data elements other than power xpert blue

These style issues were actually being inherited from the "Responsive Table" List example.  I've fixed the style bleed so we don't get any unwanted hover effects now.

**Comment by emclaug2** [comment]
> Looks good 👍 .
> 
> Few comments from my side:
> 
> 1. Note is missing below table `Remember that in a real application ...`
> 2. Table width is not matching with design.
> 3. In mobile color of `#1, #2 and #3` is gray-500
> 4. Can we remove divider prop from last column?


I've added the missing text, adjusted the width, updated the index color of list view, and made the last divider in the list view full-width.

**Comment by emclaug2** [comment]
> Number cells shall be right-aligned. Text cells shall stay left-aligned
You got it, should be good now. 

> What is this white space on the bottom for?
It's to keep you on your toes.  It's been removed :)

**Review by jeffvg — APPROVED** [review]
looks good

File: src/pages/form-validation/table/index.tsx [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+} from '@material-ui/core';
+import { Menu } from '@material-ui/icons';
+import { makeStyles, Theme, useTheme } from '@material-ui/core/styles';
+import { useDispatch } from 'react-redux';
+import * as Colors from '@pxblue/colors';
+import { TOGGLE_DRAWER } from '../../../redux/actions';
+
+const useStyles = makeStyles((theme: Theme) => ({
+    toolbarGutters: {
+        padding: '0 16px',
```
Comment (huayunh) [inline src/pages/form-validation/table/index.tsx:25]:
```suggestion
        padding: `0 ${theme.spacing(2)}px`,
```

File: src/pages/form-validation/table/index.tsx [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+export const TableFormValidation = (): JSX.Element => {
+    const theme = useTheme();
+    const classes = useStyles(theme);
+    const dispatch = useDispatch();
+
+    const getTable = (): JSX.Element => (
+        <TableContainer component={Paper} className={classes.tableContainer}>
+            <Table>
+                <TableHead style={{ fontWeight: 600 }}>
```
Comment (huayunh) [inline src/pages/form-validation/table/index.tsx:59]: may want to log an enhancement and make this into theme

File: src/pages/form-validation/table/index.tsx [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    );
+
+    const getList = (): JSX.Element => (
+        <>
+            {rows.map((row, index) => (
+                <div key={row.id} style={{ display: 'flex', flexDirection: 'column' }}>
+                    <div style={{ display: 'flex', alignItems: 'center', height: 64 }}>
+                        <Typography
+                            variant={'body1'}
+                            style={{ width: '20%', paddingLeft: theme.spacing(2), color: Colors.gray[500] }}
```
Comment (huayunh) [inline src/pages/form-validation/table/index.tsx:116]: in the future we might make this dark theme. Can we do `theme.palette.text.secondary`? Avoid citing `@pxblue/colors` directly.

File: src/pages/form-validation/table/table-validation.test.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+/* eslint-disable */
```
Comment (huayunh) [inline src/pages/form-validation/table/table-validation.test.tsx:1]: why eslint-disable here?

File: src/pages/form-validation/table/index.tsx (hunk 2 above)
Comment (emclaug2) [reply src/pages/form-validation/table/index.tsx:59]: Good callout, this style actually was already provided as the default.  No issue needed & this inline style was removed.

File: src/pages/form-validation/table/table-validation.test.tsx (hunk 4 above)
Comment (emclaug2) [reply src/pages/form-validation/table/table-validation.test.tsx:1]: I think this was a copy/paste gone wrong.  This file was already linted corrected & the comment was removed.

File: src/pages/form-validation/table/index.tsx (hunk 3 above)
Comment (emclaug2) [reply src/pages/form-validation/table/index.tsx:116]: The @pxblue/colors import has been removed for this example.

**Comment by huayunh** [comment]
## Alignment
"ID" shall be right-aligned, and "Name" shall be left-aligned
image

## Mobile List Styling
As for the mobile list view styling, I suggest this:
image

image

The idea here is that these elements are lined up
image

I would avoid relative width as much as possible — your spacing shall look correct at a minimum width of 320px.

## Inline Styles
I noticed that you have a lot of inline styles. I would suggest pull them out and put them together in the createMuiTheme function. If you do it this way, your code snippet gets less crowded, making it easier to sort out the code structure. Use inline styles only when you have substantial trouble with the `className` approach.

**Comment by emclaug2** [comment]
This demo has been refactored to use InfoListItems and mobile view should match angular now.

**Comment by ektasawant** [comment]
Changes looks good .

File: src/pages/form-validation/table/index.tsx [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+                                    },
+                                }}
+                            />
+                        }
+                    />
+                    <InfoListItem
+                        title={''}
+                        style={{ height: theme.spacing(2), marginBottom: -theme.spacing(2) }}
+                        divider={index === rows.length - 1 ? 'full' : 'partial'}
+                    />
```
Comment (huayunh) [inline src/pages/form-validation/table/index.tsx:159]: Feels hacky.... you introduced a negative margin + a blank ILI to do the divider. The end result looks visually good though. Wonder how @joebochill feels about this.

**Review by huayunh — COMMENTED** [review]
> ## Alignment
>"ID" shall be right-aligned, and "Name" shall be left-aligned
> image

File: src/pages/form-validation/table/index.tsx (hunk 5 above)
Comment (emclaug2) [reply src/pages/form-validation/table/index.tsx]: Ok so the negative margin wasn't required; but the 16px InfoListItem is still there just so we don't have to do any calculations to figure out how big a partial divider should be.

File: src/pages/form-validation/table/index.tsx [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+                                    },
+                                }}
+                            />
+                        }
+                    />
+                    <InfoListItem
+                        title={''}
+                        style={{ height: theme.spacing(2) }}
+                        divider={index === rows.length - 1 ? 'full' : 'partial'}
+                    />
```
Comment (joebochill) [inline src/pages/form-validation/table/index.tsx:159]:
```suggestion
                    <Divider variant={'inset'} style={{marginTop: theme.spacing(1), marginBottom: theme.spacing(1)}} />
```

**Review by daileytj — APPROVED** [review]

**Pull request merged**